code/datums/emote.dm

```diff
@@ -480,8 +483,12 @@
 
 	if(ishuman(user))
 		var/mob/living/carbon/human/H = user
-		if(species_type_whitelist_typecache && H.dna && !is_type_in_typecache(H.dna.species, species_type_whitelist_typecache))
-			return FALSE
+		if(H.dna)
+			if(!is_type_in_typecache(H.dna?.species, species_type_whitelist_typecache))
+				return FALSE
+
+			if(is_type_in_typecache(H.dna?.species, species_type_blacklist_typecache))
+				return FALSE
 
```

Comment:
you don't need the `?` because if `dna` is null, `dna.species` will not be checked due to the aforementioned check

Reply:
oh you're totally right, my bad. Got the problem backwards in my head.